sncast_std/src/lib.cairo

```diff
@@ -64,3 +64,59 @@ fn declare(contract_name: felt252, max_fee: Option<felt252>) -> DeclareResult {
 
     DeclareResult { class_hash, transaction_hash }
 }
+
+#[derive(Drop, Clone)]
+struct DeployResult {
+    contract_address: ContractAddress,
+    transaction_hash: felt252,
+}
+
+fn deploy(
+    class_hash: ClassHash,
+    constructor_calldata: Array::<felt252>,
+    salt: Option<felt252>,
+    unique: bool,
+    max_fee: Option<felt252>
+) -> DeployResult {
+    let class_hash_felt: felt252 = class_hash.into();
+    let mut inputs = array![class_hash_felt];
+
+    let calldata_len = constructor_calldata.len();
+    inputs.append(calldata_len.into());
+
+    let mut i = 0;
+    loop {
+        if i == calldata_len {
+            break;
+        }
+        inputs.append(*constructor_calldata[i]);
+        i += 1;
+    };
+
+    match salt {
+        Option::Some(val) => {
+            inputs.append(0);
+            inputs.append(val);
+        },
+        Option::None => inputs.append(1),
+    };
+
+    inputs.append(unique.into());
+
+    match max_fee {
+        Option::Some(val) => {
+            inputs.append(0);
+            inputs.append(val);
+        },
+        Option::None => inputs.append(1),
+    };
```

Comment:
let's use `option_as_tuple` and `join_arrays` for input serialisation
https://github.com/foundry-rs/starknet-foundry/blob/master/snforge_std/src/cheatcodes/tx_info.cairo#L38-L52